src/backend/sev/data.rs

```diff
@@ -5,10 +5,53 @@ pub use crate::backend::kvm::data::{dev_kvm, kvm_version};
 use crate::backend::probe::x86_64::{CpuId, Vendor};
 use crate::backend::Datum;
 
+use crate::backend::sev::snp::vcek::{get_vcek_reader_with_path, sev_cache_dir};
+use colorful::core::StrMarker;
 use std::arch::x86_64::__cpuid_count;
 use std::fs::OpenOptions;
 use std::mem::MaybeUninit;
 
+pub fn has_vcek_cache() -> Datum {
+    let name = "SEV-SNP VCEK key cache file".to_str();
+
+    let cache_dir = match sev_cache_dir() {
+        Ok(path) => path,
+        Err(e) => {
+            return Datum {
+                name,
+                pass: false,
+                info: Some(e.to_string()),
+                mesg: Some("enarx expects `/var/cache/amd-sev` to exist".to_str()),
+            }
+        }
+    };
+
+    let path = match get_vcek_reader_with_path(cache_dir).map(|(p, _)| p) {
+        Err(e) => {
+            return Datum {
+                name,
+                pass: false,
+                info: Some(e.to_string()),
+                mesg: Some("enarx expects `/var/cache/amd-sev` to exist".to_str()),
```

Comment:
Shouldn't this error be different from the one above? If not, then why not merge this into a single "pipe" with a `map`?

Reply:
fixed